From 0411ab6f03d6bfe110a2ead2347f0d3bd9d9d921 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 27 Aug 2024 08:15:36 +0300
Subject: [PATCH] [FSW-8694] fs1: retrieve spi efuse sector offset

Retrieving spi efuse images is different from
firmware image retrieving method.
Get boot desc function used to identify spi
efuse sector it has sub manifest type, which
contains spi efuse related manifest, image
descriptors, and efuse binaries.

Change-Id: I79243d8435652995ce1228cf12ead29a880a0d8d
---

diff --git a/common/boot_flow.c b/common/boot_flow.c
index d82b61b..6aaf251 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -270,3 +270,13 @@
 	boot_rom_status_update(PROGRESS_BOOT_RECOVERY_IMG_IMG0_FINISH);
 	return BOOT_ROM_SUCCESS;
 }
+
+uint32_t boot_flow_get_spi_efuse_manifest_offset(enum hl_fw_manifest_type manifest_idx)
+{
+	struct boot_desc desc;
+
+	if (boot_desc_get_img_desc(manifest_idx, HL_FW_IMG_FAKE_EFUSE, &desc, 0) == BOOT_ROM_FAIL)
+		return 0;
+
+	return desc.spi_offset;
+}
diff --git a/common/boot_flow.h b/common/boot_flow.h
index 28b7bca..9f8554b 100644
--- a/common/boot_flow.h
+++ b/common/boot_flow.h
@@ -20,5 +20,6 @@
 uint32_t boot_rom_boot_flow_load_recovery_images(enum boot_rom_manifests recovery_manifest,
 						  uint32_t img_count,
 						  enum boot_rom_progress img_sts_base);
+uint32_t boot_flow_get_spi_efuse_manifest_offset(enum hl_fw_manifest_type manifest_idx);
 
 #endif /* _BOOT_FLOW_H_ */
